langtest/datahandler/datasource.py

```diff
@@ -876,3 +876,292 @@ def _row_to_sample_classification(self, data_row: Dict[str, str]) -> Sample:
             original=original,
             expected_results=SequenceClassificationOutput(predictions=[label]),
         )
+
+
+class CustomCSVDataset(_IDataset):
+    """
+    A class to handle CSV files datasets. Subclass of _IDataset.
```

Comment:
How is that different from the `CSVDataset` we already have?

Reply:
I think `CSVDataset` is dynamically called based on the  .csv extenstion that we have in the file path 
you can check the `load` method of `DataFactory`
for custom column names we are using dictionary format in data so it will not match .

Reply:
I would rather have only one class that handles CSV files

Reply:
we can try that